internal/venice-common/src/main/java/com/linkedin/venice/kafka/consumer/ApacheKafkaConsumer.java

```diff
@@ -117,18 +139,22 @@ public void unSubscribe(PubSubTopicPartition pubSubTopicPartition) {
       if (topicPartitionList.remove(topicPartition)) {
         kafkaConsumer.assign(topicPartitionList);
       }
-      assignments.remove(pubSubTopicPartition);
+      assignments.remove(topicPartition);
     }
     topicPartitionsOffsetsTracker
         .ifPresent(partitionsOffsetsTracker -> partitionsOffsetsTracker.removeTrackedOffsets(topicPartition));
   }
 
   @Override
   public void batchUnsubscribe(Set<PubSubTopicPartition> pubSubTopicPartitionSet) {
-    Set<TopicPartition> newTopicPartitionAssignment = new HashSet<>(kafkaConsumer.assignment());
-    newTopicPartitionAssignment.removeAll(convertPubSubTopicPartitionSetToTopicPartitionSet(pubSubTopicPartitionSet));
-    kafkaConsumer.assign(newTopicPartitionAssignment);
-    assignments.removeAll(pubSubTopicPartitionSet);
+    pubSubTopicPartitionSet.forEach(
+        pubSubTopicPartition -> assignments.remove(
+            new TopicPartition(
+                pubSubTopicPartition.getPubSubTopic().getName(),
+                pubSubTopicPartition.getPartitionNumber())));
+    Collection<TopicPartition> kafkaTopicPartitions =
+        convertPubSubTopicPartitionsToTopicPartitions(assignments.values());
```

Comment:
Hmm... that looks a bit weird. Why call `convertPubSubTopicPartitionsToTopicPartitions` if you have all of the objects you need ready to go in `assignments.keySet()`? Unless perhaps you didn't intend to keep `assignments` as a map?

Reply:
Yeah, you are right. We do not need this `convertPubSubTopicPartitionsToTopicPartitions` anymore.